pykechain/client.py

```diff
@@ -1105,6 +1105,106 @@ def create_activity(
+        to clone and a dict of new values to assign, e.g. `{activity.id: {"name": "Cloned activity"}}`
+        :type activity_update_dicts: dict
+        :param clone_parts: (O) whether to clone the data models configured in the activities, defaults to False
+        :type clone_parts: bool
+        :param clone_part_instances: (O) whether to clone the part instances of the data model configured in the
+            activities, defaults to True
+        :type clone_part_instances: bool
+        :param clone_children: (O) whether to clone child parts
+        :type clone_children: bool
+        :param excluded_parts: (O) list of Part2 objects or UUIDs to exclude from being cloned,
+            maintaining the original configuration of the widgets.
+        :type excluded_parts: list
+        :param part_parent_model: (O) parent Part2 object or UUID for the copied data model(s)
+        :type part_parent_model: Part2
+        :param part_parent_instance: (O) parent Part2 object or UUID for the copied part instance(s)
+        :type part_parent_instance: Part2
+        :param part_model_rename_template: (O) renaming template for part models. Must contain "{name}"
+        :type part_model_rename_template: str
+        :param part_instance_rename_template: (O) renaming template for part instances. Must contain "{name}"
+        :type part_instance_rename_template: str
+        :param asynchronous: If true, immediately returns without activities (default = False)
+        :type asynchronous: bool
+        :return: list of cloned activities
+        :rtype: list
+        :raises APIError if cloned
+        """
+        update_name = 'activity_update_dicts'
+        activity_ids = check_list_of_base(activities, cls=Activity2, key='activities')
+        update_dicts = check_type(activity_update_dicts, dict, key=update_name) or dict()
+
+        if not all(key in activity_ids for key in update_dicts.keys()):
+            incorrect_ids = [key for key in update_dicts.keys() if key not in activity_ids]
+            raise IllegalArgumentError(
+                "The `{}` must contain updates to activities that are to be cloned. "
+                "Did not recognize the following UUIDs:\n{}".format(update_name, '\n'.join(incorrect_ids)))
+
+        elif not all(isinstance(value, dict) for value in update_dicts.values()):
+            raise IllegalArgumentError("The `{}` must be a dict of dicts.".format(update_name))
+
+        activities = [dict(id=uuid, **update_dicts.get(uuid, {})) for uuid in activity_ids]
+
+        data = dict(
+            parent_id=check_base(parent, cls=Activity2, key='parent'),
+            clone_parts=check_type(clone_parts, bool, 'clone_parts'),
+            clone_part_instances=check_type(clone_part_instances, bool, 'clone_part_instances'),
+            clone_children=check_type(clone_children, bool, 'clone_children'),
+            exclude_model_ids=check_list_of_base(excluded_parts, cls=Part2, key='excluded_models') or [],
```

Comment:
here you use `kwargs`, rest `args`. Maybe consider ironing that out (accept and commit the proposed change) for consistency sake.
```suggestion
            exclude_model_ids=check_list_of_base(excluded_parts, Part2, 'excluded_models') or [],
```